subprojects/launcher/src/integTest/groovy/org/gradle/launcher/daemon/DaemonFeedbackIntegrationSpec.groovy

```diff
@@ -175,8 +175,10 @@ task sleep {
         then:
         getLogs(executer.daemonBaseDir).size() == 0 //we should connect to the foreground daemon so no log was created
 
-        daemon.standardOutput.count(DaemonMessages.ABOUT_TO_START_RELAYING_LOGS) == 0
-        daemon.standardOutput.count("info me!") == 1
+        poll(5) {
```

Comment:
I think we're missing a flush at the end of a build. We should guarantee that every log event generated during a build is forwarded to the logging outputs _before_ we return the "finished the build" result to the client, otherwise we'll likely see some weird things happening.

We shouldn't have to change this test as part of this work, I think.

Reply:
I've rolled this back and set it up so that we flush the output before we either add or remove a listener so we ensure that listeners receive all of the notifications that they should and only the notifications that they should.  This results in a flush at the end of a build right before we disconnect the listener sending events back to the daemon client.

If we want a stronger guarantee than that, we can pull `flush()` up to `LoggingOutputInternal` and make an explicit call at the end of a build.

Otherwise, I've rolled back the changes to this test.